setup_rucio.py

```diff
@@ -16,190 +16,92 @@
+# - Matt Snyder <msnyder@bnl.gov>, 2019
 # - Ruturaj Gujar <ruturaj.gujar23@gmail.com>, 2019
-# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020
+# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020-2021
 
 import glob
 import os
-import re
-import subprocess
+import shutil
 import sys
 
-from distutils.command.sdist import sdist as _sdist
+from setuptools import setup, find_packages
 
-if sys.version_info < (2, 4):
-    print('ERROR: Rucio requires at least Python 2.5 to run.')
-    sys.exit(1)
-
-sys.path.insert(0, os.path.abspath('lib/'))
+from setuputil import server_requirements_table, match_define_requirements, run_shell_command, get_rucio_version
 
-from rucio import version  # noqa
+if sys.version_info < (3, 6):
+    print('ERROR: Rucio Server requires at least Python 3.6 to run.')
+    sys.exit(1)
 
-try:
-    from setuptools import setup, find_packages
-except ImportError:
-    from ez_setup import use_setuptools
-    use_setuptools()
+install_requires, extras_require = match_define_requirements(server_requirements_table)
 
 name = 'rucio'
-packages = find_packages('lib/')
+packages = find_packages(where='lib')
 description = "Rucio Package"
-IsRelease = False
-requirements_files = ['etc/pip-requires', 'etc/pip-requires-client']
-data_files = [('rucio/etc/', glob.glob('etc/*.template') + ['etc/pip-requires', 'etc/pip-requires-client',
-                                                            'etc/pip-requires-test']),
-              ('rucio/etc/web', glob.glob('etc/web/*.template')),
-              ('rucio/tools/', ['tools/bootstrap.py', 'tools/reset_database.py']),
-              ('rucio/etc/mail_templates/', glob.glob('etc/mail_templates/*.tmpl'))]
-
+data_files = [
+    ('rucio/', ['requirements.txt']),
+    ('rucio/etc/', glob.glob('etc/*.template')),
+    ('rucio/etc/web', glob.glob('etc/web/*.template')),
+    ('rucio/tools/', ['tools/bootstrap.py', 'tools/reset_database.py']),
+    ('rucio/etc/mail_templates/', glob.glob('etc/mail_templates/*.tmpl')),
+]
 scripts = glob.glob('bin/rucio*')
 
-copy_args = sys.argv[1:]
-if '--release' in copy_args:
-    IsRelease = True
-    copy_args.remove('--release')
+if os.path.exists('build/'):
+    shutil.rmtree('build/')
+if os.path.exists('lib/rucio_clients.egg-info/'):
+    shutil.rmtree('lib/rucio_clients.egg-info/')
+if os.path.exists('lib/rucio.egg-info/'):
+    shutil.rmtree('lib/rucio.egg-info/')
 
 # Flags to know if the installation is done through pip and against git
 using_pip = os.path.basename(os.path.dirname(__file__)).startswith('pip-')
 using_git = os.path.isdir('.git')
 
 
-def run_git_command(cmd):
-    """
-    Run a git command in path and return output"
-
-    :param cmd: the git command.
-    :return: Output of the git command.
-    """
-    output = subprocess.Popen(["/bin/sh", "-c", cmd], stdout=subprocess.PIPE)
-    return output.communicate()[0].strip()
-
-
 if using_pip and using_git:
+    # FIXME: duplicate in tools/generate_version.py
```

Comment:
I wonder if we still need this?
The vcversion file I usually create outside of the procedure (with the `tools/generate_version.py`) anyway. So that part doesn't seem useful anymore.

Reply:
I leave that for you to decide. If you dont need it here, I can remove it now

Reply:
Let's remove it then, I think it's just clutter. Thanks :-)